internal/cssc/cssc.go

```diff
@@ -0,0 +1,190 @@
+	if !strings.Contains(filterPolicy, ":") {
+		return Filter{}, errors.New("filter-policy should be in the format repo:tag")
+	}
+	repoTag := strings.Split(filterPolicy, ":")
+	filterRepoName := repoTag[0]
+	filterRepoTagName := repoTag[1]
+
+	// Connect to the remote repository
+	repo, err := remote.NewRepository(fmt.Sprintf("%s/%s", loginURL, filterRepoName))
+	if err != nil {
+		panic(err)
+	}
+	repo.Client = &auth.Client{
+		Client: retry.DefaultClient,
+		Cache:  auth.NewCache(),
+		Credential: auth.StaticCredential(loginURL, auth.Credential{
+			Username: username,
+			Password: password,
+		}),
+	}
+
+	// Get manifest and read content
+	_, pulledManifestContent, err := oras.FetchBytes(ctx, repo, filterRepoTagName, oras.DefaultFetchBytesOptions)
+	if err != nil {
+		return Filter{}, errors.Wrap(err, "error fetching manifest content when reading the filter policy")
+	}
+	var pulledManifest v1.Manifest
+	if err := json.Unmarshal(pulledManifestContent, &pulledManifest); err != nil {
+		panic(err)
+	}
+	var fileContent []byte
+	for _, layer := range pulledManifest.Layers {
+		fileContent, err = content.FetchAll(ctx, repo, layer)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Unmarshal the JSON file data to Filter struct
+	var filter = Filter{}
+	if err := json.Unmarshal(fileContent, &filter); err != nil {
+		return Filter{}, errors.Wrap(err, "error unmarshalling json content when reading the filter policy")
+	}
+	return filter, nil
+}
+
+// Reads the filter json from the specified file path and returns the Filter struct
+func GetFilterFromFilePath(filePath string) (Filter, error) {
+	file, err := os.ReadFile(filePath)
+	if err != nil {
+		return Filter{}, errors.Wrap(err, "error reading the filter json file from file path")
+	}
+
+	var filter = Filter{}
+	if err := json.Unmarshal(file, &filter); err != nil {
+		return Filter{}, errors.Wrap(err, "error unmarshalling json content when reading the filter file from file path")
+	}
+
+	return filter, nil
+}
+
+// Applies filter to filter out the repositories and tags from the ACR according to the specified criteria and returns the FilteredRepository struct
+func ApplyFilterAndGetFilteredList(ctx context.Context, acrClient api.AcrCLIClientInterface, filter Filter) ([]FilteredRepository, error) {
+	var filteredRepos []FilteredRepository
+
+	for _, filterRepo := range filter.Repositories {
+		if filterRepo.Enabled != nil && !*filterRepo.Enabled {
+			continue
+		}
+		if filterRepo.Repository == "" || filterRepo.Tags == nil || len(filterRepo.Tags) == 0 || filterRepo.Tags[0] == "" {
+			continue
+		}
+
+		tagList, err := tag.ListTags(ctx, acrClient, filterRepo.Repository)
+		if err != nil {
+			if strings.Contains(err.Error(), "404") {
+				continue
+			}
+			return nil, err
+		}
+
+		if len(filterRepo.Tags) == 1 && filterRepo.Tags[0] == "*" {
+			for _, tag := range tagList {
+				if strings.Contains(*tag.Name, "-patched") {
+					originalTag := strings.Split(*tag.Name, "-patched")[0]
```

Comment:
Nit. Since -patched is a suffix a more efficient approach would be to get a substring from [0 : len(str)- len("-patched")-1] to avoid the string search

Reply:
Updated as per your suggestion